src/components/Login/LoginContainer.tsx

```diff
@@ -144,17 +145,26 @@ export const InitialisationContainer: FunctionComponent<NavigationProps> = ({
     }
   }, [isLoading, endpoint, navigation, token, features]);
 
+  const setState = useState()[1];
   useEffect(() => {
-    const skipScanningIfParamsInDeepLink = async (): Promise<void> => {
+    const skipScanningIfParamsInDeepLink = async (): Promise<any> => {
       const { queryParams } = await Linking.parseInitialURLAsync();
+      const queryEndpoint = queryParams?.endpoint;
+      if (queryEndpoint && !RegExp(DOMAIN_CHECK).test(queryEndpoint)) {
+        const error = new Error(`Invalid endpoint: ${queryEndpoint}`);
+        Sentry.captureException(error);
+        setState(() => {
+          throw error;
+        });
```

Comment:
Actually, come to think of it, maybe an alert will suffice, like in the normal case, what do you think?

Reply:
hahaha, hmm it depends. 

If we do the error boundary, it may be better cause at least user can see that something went wrong haha.

But an alert works fine too~ Maybe we stick with this first? :O

Reply:
Yea am good with either!

Reply:
ok~